project-addons/custom_partner/partner.py

```diff
@@ -174,6 +174,23 @@ def _calculate_annual_invoiced(self):
                     'monthly_invoiced': monthly_invoiced, 'past_month_invoiced': past_month_invoiced}
             partner.write(vals)
 
+    @api.model
+    def _unblock_invoices(self):
+        date_limit = date.today() - timedelta(days=7)
+        payment_term_ids = self.env['account.payment.term'].search([('blocked', '=', 'True')])
+        partner_ids = self.env['res.partner'].search([('property_payment_term', 'in', payment_term_ids.ids)])
+        invoice_ids = self.env['account.invoice'].search([('date_due', '<=', date_limit),
+                                                          ('state', '=', 'open'),
+                                                          ('partner_id', 'in', partner_ids.ids),
+                                                          ('number', 'not like', '%_ef%'),
+                                                          ('number', 'not like', 'VEN%')])
+        move_line_obj = self.env['account.move.line']
+        for invoice in invoice_ids:
+            move_line = move_line_obj.search([('stored_invoice_id', '=', invoice.id),
```

Comment:
Te ahorrarías el for si hace 'store_invoice_id', 'in', invoice_ids.ids

Reply:
Pero si hago esto, me devolvería un array de move_lines y luego tengo que escribir sobre cada uno de ellos y no podría hacerlo directamente no?

Reply:
No, puedes ecribir sobre todo el array.

Reply:
Okey, lo modifico y lo pruebo